### PR TITLE
#58 - add @spatie laravel-permission tools

### LINKED ISSUE #58: Add spatie/laravel-permission tools

### PR DESCRIPTION
**Pull request by JakubKermes**
This should close #58.
Added options for adding roles and permissions to users, or other objects in database.

### CODE REVIEW

**Comment by JakubKermes** [comment]
This should be working already, but needs eloquent tools to actually implement this into tests, waiting for merge of #42.

File: src/Features/Traits/Optional/SpatiePermission.php [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    /**
+     * @When :objectName can have role
+     * @When user can have role
+     * @throws BindingResolutionException
+     */
+    public function objectCanHaveRoles(string $objectName = "User"): void
+    {
+        $objectClass = ContextHelper::getClassHelper()->recognizeObjectClass($objectName);
+        $used_traits = class_uses_recursive($this->getContainer()->make($objectClass));
+        Assert::assertTrue(in_array("Spatie\Permission\Traits\HasRoles", $used_traits, true));
```
Comment (krzysztofrewak) [inline src/Features/Traits/Optional/SpatiePermission.php:26]:
```suggestion
        $usedTraits = class_uses_recursive($this->getContainer()->make($objectClass));
        Assert::assertTrue(in_array("Spatie\Permission\Traits\HasRoles", $usedTraits, strict: true));
```

Comment (krzysztofrewak) [reply src/Features/Traits/Optional/SpatiePermission.php:26]: Plus you could move this class reference to some other file like here: https://github.com/blumilksoftware/blt/blob/main/src/LaravelContracts.php

**Review by krzysztofrewak — CHANGES_REQUESTED** [review]
If #70 would be merged before this, you should add some documentation as well.

**Comment by JakubKermes** [comment]
There is documentation in #70 already for this

**Pull request merged**